### ISSUE TITLE
MaskedInput accessibility

### ISSUE THREAD

**Issue by qroll** [LifeSG/react-design-system#884]
**Component**
`MaskedInput`

**Describe the feature request**
- Align the MaskedInput with the accessibility behaviour defined in the Figma specs
- You may refer to the implementation in `UneditableSection.Item`

**Acceptance criteria**
- Roles should be correctly read out by the AT
- Masked input should be described correctly
- Label and error should be correctly associated with the input field
- Disabled, readonly state should be correctly read out by the AT
- Should remain focusable when disabled
- Mask toggle button should be correctly handled by the AT
- Minimally test in NVDA + Chrome (Windows), VO + Safari (macOS), Talkback + Chrome (Android)

**References**
[Figma](https://www.figma.com/design/wDAxbRmlq4xOn4UJ8zPuEF/%F0%9F%A7%91%F0%9F%8F%BB%E2%80%8D%F0%9F%A6%AF-Flagship-A11y-Annotations?node-id=1996-2467&p=f&m=dev)
[`UneditableSection.Item` reference](https://github.com/LifeSG/react-design-system/blob/master/src/uneditable-section/section-item.tsx)
